setup/fusion/scripts/Comp/colorbleed/switch_ui.py

```diff
@@ -146,16 +148,20 @@ def _refresh(self):
 
     def _on_switch(self):
 
+        asset = self._assets.currentText()
+        _comp = avalon.fusion.get_current_comp()
         if not self._use_current.isChecked():
             file_name = self._comps.itemData(self._comps.currentIndex())
+            # Get the current comp's App (FusionUI)
+            _fusion = _comp.GetApp()
+            # Open the selected comp
+            _fusion.LoadComp(file_name)
```

Comment:
This might fail (e.g. file is invalid), ensure it returns the newly loaded comp.

Reply:
This won't be the case as the filename is derived from browsing to the file, see `_on_open_from_dir`

Reply:
> This won't be the case as the filename is derived from browsing to the file, see _on_open_from_dir

The file could change after browsing, plus the file itself could be in invalid working comp. The result will be that the switch command will be switching the wrong comp around - which is dangerous. Right?

Reply:
> plus the file itself could be in invalid working comp

Define an invalid working comp?

I can add `os.path.isfile` to ensure the file is not deleted in the meantime. 

>Furthermore, if the file is invalid it will show a popup message too! ;) Which you might want to suppress?

I would like Fusion to give me a pop up message if the comp file is invalid, I can add a filter to find only `.comp` files,

Reply:
> I would like Fusion to give me a pop up message if the comp file is invalid, I can add a filter to find only .comp files,

Save an empty file name `test.comp`, try to open it.

```python
fusion.LoadComp("C:/test.comp")
print("A")
```

The "A" will be printed after you've clicked the pop-up. It will continue onwards in your script running on the wrong composition.

I'm testing with an empty file, but it could just as well be any comp file that didn't save correctly.